src/index.ts

```diff
@@ -73,64 +74,58 @@ class Index<T> extends MeiliAxiosWrapper implements Types.IndexInterface<T> {
     }
-    if (options !== undefined) {
-      if (options.offset !== undefined) {
-        params.offset = options.offset
-      }
-      if (options.limit !== undefined) {
-        params.limit = options.limit
-      }
-      if (options.attributesToRetrieve !== undefined) {
-        if (Array.isArray(options.attributesToRetrieve)) {
-          params.attributesToRetrieve = options.attributesToRetrieve.join(',')
-        } else {
-          params.attributesToRetrieve = options.attributesToRetrieve
-        }
+    if (method.toUpperCase() === 'POST') {
+      return await this.post(url, removeUndefinedFromObject(params), {
+        cancelToken: this.cancelTokenSource.token,
+      })
+    } else if (method.toUpperCase() === 'GET') {
+      const getParams: Types.GetSearchRequest = {
+        ...params,
+        facetFilters:
+          Array.isArray(options?.facetFilters) && options?.facetFilters
+            ? JSON.stringify(options.facetFilters)
+            : undefined,
+        facetsDistribution: options?.facetsDistribution
+          ? JSON.stringify(options.facetsDistribution)
+          : undefined,
+        attributesToRetrieve: options?.attributesToRetrieve
+          ? options.attributesToRetrieve.join(',')
+          : undefined,
+        attributesToCrop: options?.attributesToCrop
+          ? options.attributesToCrop.join(',')
+          : undefined,
+        attributesToHighlight: options?.attributesToHighlight
+          ? options.attributesToHighlight.join(',')
+          : undefined,
       }
 
-      if (options.attributesToCrop !== undefined) {
-        if (Array.isArray(options.attributesToCrop)) {
-          params.attributesToCrop = options.attributesToCrop.join(',')
-        } else {
-          params.attributesToCrop = options.attributesToCrop
-        }
-      }
-      if (options.cropLength !== undefined) {
-        params.cropLength = options.cropLength
-      }
-      if (options.attributesToHighlight !== undefined) {
-        if (Array.isArray(options.attributesToHighlight)) {
-          params.attributesToHighlight = options.attributesToHighlight.join(',')
-        } else {
-          params.attributesToHighlight = options.attributesToHighlight
-        }
-      }
-      if (options.filters !== undefined) {
-        params.filters = options.filters
-      }
-      if (options.matches !== undefined) {
-        params.matches = options.matches
-      }
-      if (options.facetFilters !== undefined) {
-        params.facetFilters = JSON.stringify(options.facetFilters)
-      }
-      if (options.facetsDistribution !== undefined) {
-        params.facetsDistribution = JSON.stringify(options.facetsDistribution)
-      }
+      return await this.get(url, {
+        params: removeUndefinedFromObject(getParams),
+        cancelToken: this.cancelTokenSource.token,
+      })
+    } else {
+      throw new MeiliSearchError(
+        'method parameter should be either POST or GET'
+      )
```

Comment:
Cannot Typscript handle this part? 🙂

Reply:
Well, yes, but not if the user is not using typescript but javascript. In which case I prefer the error to be really obvious.